ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncAlgorithm.java

```diff
@@ -106,6 +106,7 @@ private CompletableFuture<Void> handleSyncStep(final Hash firstHash) {
     executeSyncStep(firstHash)
         .whenComplete(
             (result, error) -> {
+              context.getBackwardChain().removeFromHashToAppend(firstHash);
```

Comment:
Why does the `context.getBackwardChain().removeFromHashToAppend` need to be done here? This is done in the `handleSyncStepSuccess` and in `handleSyncStepError` already.